include/exadg/utilities/spatial_aware_function.h

```diff
@@ -0,0 +1,62 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2024 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef EXADG_AERO_UTILITIES_SPATIAL_AWARE_FUNCTION_H_
+#define EXADG_AERO_UTILITIES_SPATIAL_AWARE_FUNCTION_H_
+
+#include <deal.II/base/function.h>
+
+namespace ExaDG
+{
+namespace Utilities
+{
+/**
+ * This is a @c dealii::Function, that provides information if its result is constant over space at
```

Comment:
"constant", while below you write "varies_in_space". I would unify this.

Reply:
Please add the following documentation here:

The intention behind is to bypass a potentially expensive evaluation of `dealii::Function::value()` (which the internal implementation will call for each quadrature point) in case one knows a priori that the function does not vary in space.

Reply:
[4776f56](https://github.com/exadg/exadg/pull/644/commits/4776f5622e28a7d092692b74ed7fe858ab539238)